cachito/workers/scm.py

```diff
@@ -62,15 +63,50 @@ def _reset_git_head(self, repo):
                 f'of "{self.ref}" is valid.'
             )
 
+    def _verify_archive(self):
+        """
+        Verify the archive containing the git repository.
+
+        :raises CachitoError: if 'git fsck' fails for the extracted sources
+        """
+        log.debug("Verifying the archive at %s", self.sources_dir.archive_path)
+        if not os.path.exists(self.sources_dir.archive_path) or not tarfile.is_tarfile(
+            self.sources_dir.archive_path
+        ):
+            raise CachitoError(f"No valid tar archive at {self.sources_dir.archive_path}")
+
+        with tempfile.TemporaryDirectory(prefix="cachito-") as temp_dir:
+            with tarfile.open(self.sources_dir.archive_path, mode="r:gz") as tar:
+                tar.extractall(temp_dir)
+
+            cmd = ["git", "fsck"]
+            repo_path = os.path.join(temp_dir, "app")
+            try:
+                subprocess.run(cmd, cwd=repo_path, check=True)
+            except subprocess.CalledProcessError as exc:
+                log.error(
+                    "Cachito found an error when verifying the generated archive at '%s': %s",
+                    self.sources_dir.archive_path,
+                    exc,
+                )
+                raise CachitoError(f"Invalid tar archive at {self.sources_dir.archive_path!s}")
```

Comment:
Same as above if you decide to use just the "archive" for user facing error.

Reply:
Done